packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -0,0 +1,509 @@
+import { Events as EVENTS, ViewportType } from '../enums';
+import {
+  IVideoViewport,
+  VideoViewportProperties,
+  Point3,
+  Point2,
+} from '../types';
+import { Transform } from './helpers/cpuFallback/rendering/transform';
+import renderingEngineCache from './renderingEngineCache';
+import { triggerEvent } from '../utilities';
+import Viewport from './Viewport';
+import { getOrCreateCanvas } from './helpers';
+
+export type VideoCamera = {
+  pan: Point2;
+  parallelScale: number;
+};
+
+export type IVideoCamera = {
+  pan?: Point2;
+  parallelScale?: number;
+};
+
+export type ViewportInput = {
+  id: string;
+  renderingEngineId: string;
+  type: ViewportType;
+  element: HTMLDivElement;
+  sx: number;
+  sy: number;
+  sWidth: number;
+  sHeight: number;
+  defaultOptions: any;
+  canvas: HTMLCanvasElement;
+};
+
+/**
+ * An object representing a single stack viewport, which is a camera
+ * looking into an internal scene, and an associated target output `canvas`.
+ */
+class VideoViewport extends Viewport implements IVideoViewport {
+  public static readonly useCustomRenderingPipeline = true;
+
+  // Viewport Data
+  readonly uid;
+  readonly renderingEngineId: string;
+  readonly canvasContext: CanvasRenderingContext2D;
+  private videoElement?: HTMLVideoElement;
+  private videoWidth = 0;
+  private videoHeight = 0;
+
+  private loop = false;
+  private mute = true;
+  private isPlaying = false;
+  private scrollSpeed = 1;
+  private fps = 30; // TODO We need to find a good solution for this.
+  private videoCamera: VideoCamera = {
```

Comment:
This is not compatible with other viewports we currently have. Therefore, it is unclear how to handle this situation. Ideally, we should use focalPoint and position, just like we do with other elements. I am confident that if we don't address this now, it will cause issues again when working with annotation and segmentation tools.